src/parser/youtube_parser.py

```diff
@@ -0,0 +1,160 @@
+            self.logger.log_error(f"An error occurred while parsing the playlist: {str(e)}")
+            return None
+
+    def parse_video(self, video_id: str) -> pd.DataFrame:
+        """
+        Parsing information about video
+        Args:
+            video_id (str):  video ID to retrieve data for.
+        Returns:
+            pandas.DataFrame: A DataFrame containing the following columns:
+                video_id (str): Video ID.
+                title (str): The title of the video.
+                published_at (str): The date and time the video was published.
+                view_count (int): The number of views the video has received.
+                like_count (int): The number of likes the video has received.
+                comment_count (int): The number of comments the video has received.
+                description (str):
+        """
+        try:
+            video_info = self._get_video_info([video_id])
+            if video_info:
+                video_data = self.create_video_dataframe(video_info)
+                return video_data
+            else:
+                return None
+
+        except Exception as e:
+            self.logger.log_error(f"An error occurred while parsing the video: {str(e)}")
+            return None
+
+    def _get_playlist_videos(self, playlist_id: str) -> list:
+        """
+        Retrieves video information from YouTube API.
+        Args:
+            playlist_id: Playlist IDs to retrieve data for.
+        Returns:
+            list: A list of dictionaries containing video information.
+        """
+        try:
+            request = self.youtube.playlistItems().list(
+                part='contentDetails',
+                playlistId=playlist_id,
+                maxResults=self.YOUTUBE_API_MAX_RESULTS
+            )
+
+            video_info = []
+            while request:
+                response = request.execute()
+                items = response.get('items', [])
+                video_ids = [item['contentDetails']['videoId'] for item in items]
+                video_info.extend(self._get_video_info(video_ids))
+                request = self.youtube.playlistItems().list_next(request, response)
+
+            return video_info
+
+        except Exception as e:
+            self.logger.log_error(f"An error occurred while retrieving playlist videos: {str(e)}")
+            return None
+
+    def _get_video_info(self, video_ids: list[str]) -> list:
+        """
+        Retrieves video information from YouTube API.
+        Args:
+            video_ids: A list of video IDs to retrieve data for.
+        Returns:
+            list: A list of dictionaries containing video information.
+        """
+        try:
+            # Execute the request for each chunk of video IDs
+            response = self.youtube.videos().list(
+                part=self.YOUTUBE_API_PARTS,
+                id=','.join(video_ids)  # Combine video IDs into a comma-separated string
```

Comment:
As a rule of thumb, if you need a comment, probably it can be extracted in a method. But, tbh, here the code is pretty self-explanatory